src/Components/Assets/AssetFilter.tsx

```diff
@@ -38,12 +32,35 @@ function AssetFilter(props: any) {
   const [asset_class, setAssetClass] = useState<string>(
     filter.asset_class || ""
   );
-  const [facilityId, setFacilityId] = useState<number | "">(filter.facility);
+  const [facilityId, setFacilityId] = useState<string | "">(filter.facility);
   const [locationId, setLocationId] = useState<string | "">(filter.location);
   const [qParams, _] = useQueryParams();
 
+  useQuery(routes.getPermittedFacility, {
+    pathParams: { id: facilityId },
+    onResponse: ({ res, data }) => {
+      if (res?.status === 200 && data) {
+        setFacility(data);
+      }
+    },
+    prefetch: !!facilityId,
+  });
+
+  useQuery(routes.getFacilityAssetLocation, {
+    pathParams: {
+      facilityId: String(facilityId),
+      locationId: String(locationId),
+    },
+    onResponse: ({ res, data }) => {
+      if (res?.status === 200 && data) {
+        setLocation(data);
+      }
+    },
+    prefetch: !!(facilityId && locationId),
+  });
+
   useEffect(() => {
-    setFacilityId(facility?.id ? facility?.id : "");
+    setFacilityId(facility?.id ? `${facility?.id}` : "");
     setLocationId(
       facility?.id === qParams.facility ? qParams.location ?? "" : ""
     );
```

Comment:
remove this useEffect and set these values in onResponse

Reply:
removing it is causing problems in with location in filter